pkg/config/config.go

```diff
@@ -346,6 +346,9 @@ func InitConfig(config Config) {
 	// compatibility with Agent5 behavior/win
 	config.BindEnvAndSetDefault("hostname_fqdn", false)
 
+	// This is used to bypass the hostname detection logic and force the EC2 instance ID as a hostname.
+	config.BindEnvAndSetDefault("hostname_force_ec2_instance_id", false)
```

Comment:
I think this may be better grouped/prefixed with the [other `ec2_` vars](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config.go#L680-L683). WDYT?

Reply:
I actually think it's more related to the hostname detection that how we use EC2.

Reply:
Good points :+1: